### PR TITLE
Fix Some Nondeterministic failures

### PR DESCRIPTION
**Pull request by DavidVorick**
This was my attempt to address some of the nondeterministic build failures that we are seeing in Travis. The code is evolving fast enough that the issues weren't really helpful, and I'm not really certain what may have been causing the issues. I've added more error information.

I think I'm in favor of closing some of the older nondeterministic build failure issues. It's good to have a log of the nondeterministic failures that travis is spitting out, but if we don't get to them quickly they become very difficult to debug.

### CODE REVIEW

**Comment by DavidVorick** [comment]
It also seems that, in general, Travis is not failing as much anymore.

**Comment by lukechampine** [comment]
poor timing on that comment lol

**Comment by DavidVorick** [comment]
It's okay both failures are fixable, wrote a commit to fix the first, working on the commit to fix the second.

**Comment by DavidVorick** [comment]
rebased, ready to go

File: api/renterhost_test.go [hunk 1]
```diff
@@ -127,10 +127,11 @@ func TestIntegrationHostAndRent(t *testing.T) {
...
-		t.Fatal("host is not reporting storage revenue")
-	}
-	if hg.FinancialMetrics.DownloadBandwidthRevenue.Cmp(types.ZeroCurrency) <= 0 {
-		t.Fatal("host is not reporting bandwidth revenue")
+	if hg.FinancialMetrics.StorageRevenue.Cmp(types.ZeroCurrency) <= 0 ||
+		hg.FinancialMetrics.DownloadBandwidthRevenue.Cmp(types.ZeroCurrency) <= 0 {
+		t.Error("Storage Revenue:", hg.FinancialMetrics.StorageRevenue)
+		t.Error("Bandwidth Revenue:", hg.FinancialMetrics.DownloadBandwidthRevenue)
+		t.Error("Full Financial Metrics:", hg.FinancialMetrics)
+		t.Fatal("Host is not displaying revenue after resolving a storage proof.")
```
Comment (lukechampine) [inline api/renterhost_test.go]: should use `t.Log` instead of `t.Error`

File: modules/consensus/synchronize_test.go [hunk 2]
```diff
@@ -1248,6 +1247,9 @@ func TestIntegrationRelaySynchronize(t *testing.T) {
 	// cst2, and then to cst3.
 	b, err := cst1.miner.AddBlock()
 	if err != nil {
+		t.Error(b.ID())
+		t.Error(cst1.cs.CurrentBlock().ID())
+		t.Error(cst2.cs.CurrentBlock().ID())
```
Comment (lukechampine) [inline modules/consensus/synchronize_test.go]: ditto

File: modules/consensus/synchronize_test.go [hunk 3]
```diff
@@ -1276,9 +1277,12 @@ func TestIntegrationRelaySynchronize(t *testing.T) {
 	// Mine a block on cst2.
 	b, err = cst2.miner.AddBlock()
 	if err != nil {
+		t.Error(b.ID())
+		t.Error(cst2.cs.CurrentBlock().ID())
+		t.Error(cst3.cs.CurrentBlock().ID())
```
Comment (lukechampine) [inline modules/consensus/synchronize_test.go]: ditto

File: modules/consensus/synchronize_test.go [hunk 4]
```diff
@@ -1301,9 +1306,12 @@ func TestIntegrationRelaySynchronize(t *testing.T) {
 	// Mine a block on cst3.
 	b, err = cst3.miner.AddBlock()
 	if err != nil {
+		t.Error(b.ID())
+		t.Error(cst1.cs.CurrentBlock().ID())
+		t.Error(cst3.cs.CurrentBlock().ID())
```
Comment (lukechampine) [inline modules/consensus/synchronize_test.go]: ditto

**Comment by DavidVorick** [comment]
fixed, rebased, build passed

**Pull request merged**